src/seedu/addressbook/Main.java

```diff
@@ -78,16 +79,33 @@ private void exit() {
 
     /** Reads the user command and executes it, until the user issues the exit command.  */
     private void runCommandLoopUntilExitCommand() {
-        Command command;
+        Command command = null;
         do {
-            String userCommandText = ui.getUserCommand();
-            command = new Parser().parseCommand(userCommandText);
-            CommandResult result = executeCommand(command);
-            recordResult(result);
-            ui.showResultToUser(result);
-
+        	try {
+                String userCommandText = ui.getUserCommand();
+                checkStorageFileExistence();
+                command = new Parser().parseCommand(userCommandText);
+                CommandResult result = executeCommand(command);
+                recordResult(result);
+                ui.showResultToUser(result);		
+        	} catch (FileDeletedException e) {
```

Comment:
Good handling!

You can consider an alternative handling too (a bit harder to code but better user experience):
- "Restore" (actually re-save) the storage file if it is missing AND inform the user that "Oh, something fishy happened and your file was missing, but we restore it and we can continue on!".